### PR TITLE
CobaltActivity copies ContentShellActivity instead of inheriting

### PR DESCRIPTION
**Pull request by johnxwork**
Avoid inheriting ContentShellActivity and copied all implementation of ContentShellActivity into CobaltActivity so that we have larger freedom modifying and customizing our own Activity.

b/372558652

### CODE REVIEW

**Review by kaidokert — APPROVED** [review]

**Comment by yell0wd0g** [comment]
QQ the deltas of content/shell/android/shell_apk/src/org/chromium/content_shell_apk/ContentShellActivity.java are just to restore it to ToT, correct? If so it'd be good to add that to the description.

File: cobalt/android/apk/app/src/main/java/dev/cobalt/coat/CobaltActivity.java [hunk 1]
```diff
@@ -33,16 +37,27 @@
...
+import org.chromium.base.library_loader.LibraryProcessType;
+import org.chromium.content_public.browser.BrowserStartupController;
+import org.chromium.content_public.browser.DeviceUtils;
+import org.chromium.content_public.browser.WebContents;
+import org.chromium.content_shell.Shell;
+import org.chromium.content_shell.ShellManager;
+import org.chromium.ui.base.ActivityWindowAndroid;
+import org.chromium.ui.base.IntentRequestTracker;
 
 /** Native activity that has the required JNI methods called by the Starboard implementation. */
```
Comment (borongc) [inline cobalt/android/apk/app/src/main/java/dev/cobalt/coat/CobaltActivity.java:59]: Should we add a comment here mentions most of the code are copied from ContentShellActivity?

Comment (borongc) [reply cobalt/android/apk/app/src/main/java/dev/cobalt/coat/CobaltActivity.java:59]: Otherwise, LGTM.

**Review by borongc — APPROVED** [review]

File: content/shell/android/shell_apk/src/org/chromium/content_shell_apk/ContentShellActivity.java [hunk 2]
```diff
@@ -73,10 +73,6 @@ protected void onCreate(final Bundle savedInstanceState) {
         mWindowAndroid.setAnimationPlaceholderView(
                 mShellManager.getContentViewRenderView().getSurfaceView());
 
-        // TODO(cobalt, b/376148547): set Chrobalt initial url and remove this function.
```
Comment (borongc) [inline content/shell/android/shell_apk/src/org/chromium/content_shell_apk/ContentShellActivity.java:76]: nit: this should be `mStartupUrl = getUrlFromIntent(getIntent());` in this line.

Comment (johnxwork) [reply content/shell/android/shell_apk/src/org/chromium/content_shell_apk/ContentShellActivity.java:76]: Thanks!

File: content/shell/android/shell_apk/src/org/chromium/content_shell_apk/ContentShellActivity.java
Comment (jellefoks) [inline content/shell/android/shell_apk/src/org/chromium/content_shell_apk/ContentShellActivity.java:1]: Now that it's Cobalt not ContentShell, shouldn't we rename it to CobaltActivity.java? Or given that that already exists, move the code into CobaltActivity?

Comment (johnxwork) [reply content/shell/android/shell_apk/src/org/chromium/content_shell_apk/ContentShellActivity.java:1]: We made a CobaltActivity under src/cobalt/android, we are going to use that moving forward and stops depending on this Chromium Activity.

File: content/shell/android/shell_apk/src/org/chromium/content_shell_apk/ContentShellActivity.java [hunk 3]
```diff
@@ -73,10 +73,8 @@ protected void onCreate(final Bundle savedInstanceState) {
...
-        if (mStartupUrl.isEmpty()) {
-            mStartupUrl = getUrlFromIntent(getIntent());
-        }
+        mStartupUrl = getUrlFromIntent(getIntent());
+
```
Comment (borongc) [inline content/shell/android/shell_apk/src/org/chromium/content_shell_apk/ContentShellActivity.java:77]: nit: it should be no extra line from the original code, we should keep it as it was

Comment (johnxwork) [reply content/shell/android/shell_apk/src/org/chromium/content_shell_apk/ContentShellActivity.java:77]: oh shoot I was looking at the upstream code: https://source.chromium.org/chromium/chromium/src/+/main:content/shell/android/shell_apk/src/org/chromium/content_shell_apk/ContentShellActivity.java;drc=62a63ab0b2ee9082b1bb25a74187bced2f599858;l=79

changing now.

**Pull request merged**